stl/inc/filesystem

```diff
@@ -1377,8 +1377,24 @@ namespace filesystem {
 #endif // ^^^ !_HAS_CXX20 ^^^
 
         _NODISCARD_FRIEND path operator/(const path& _Left, const path& _Right) { // append a pair of paths together
-            path _Tmp = _Left;
+            path _Tmp;
+            const auto _Right_size  = _Right._Text.size();
+            const auto _Right_first = _Right._Text.data();
+            const auto _Right_last  = _Right_first + _Right_size;
+
+            // Preallocate memory for the most common case
+            if (!_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {
+                // !has_root_name(_Right) && !has_root_directory(_Right)
+                const auto _Left_size               = _Left._Text.size();
+                const auto _Left_last               = _Left._Text.data() + _Left_size;
+                const bool _Left_has_trailing_slash = _Left_size && _Is_slash(_Left_last[-1]);
+
+                _Tmp._Text.reserve(_Left_size + !_Left_has_trailing_slash + _Right_size);

```

Comment:
```suggestion
                const bool _Need_adding_slash = _Left._Text.empty() || !_Is_slash(_Left._Text.back());
                _Tmp._Text.reserve(_Left_size + static_cast<size_t>(_Need_adding_slash) + _Right_size);
```
I wonder whether `_Resize_and_overwrite` (same as C++23 [`resize_and_overwrite`](https://en.cppreference.com/w/cpp/string/basic_string/resize_and_overwrite)) would be better.

Reply:
apparently there  are more changes needed to make that, can't reuse `/=` operator in this case

Reply:
I'm a bit torn about this. I wanted to avoid duplicating the actual appending logic of `/=` (as opposed to just capacity hints), but I guess in this case it could be simple enough to justify this.

Reply:
I pushed a commit that uses `_Resize_and_overwrite` and now the code has gotten a bit more complicated. To be honest, I'm not sure if it's worth it.

Reply:
Paths can be pretty long (especially if the system and application have enabled OS support for long paths), so the complexity of resize-and-overwrite seems justified. Since you've already done the excellent work here to use it, I don't think it's worth undoing in the name of implementation simplicity (it's not *too* complicated to reason about, I think).